DeepSpeech.ipynb

```diff
@@ -1145,68 +1151,66 @@
+    "    \n",
+    "    # Order the remaining items by their loss (lowest loss on top)\n",
+    "    items.sort(key=lambda a: a[3])\n",
+    "\n",
+    "    # Take only the first 10 items\n",
+    "    items = items[:10]\n",
+    "\n",
+    "    # Order this top ten items by their WER (lowest WER on top)\n",
+    "    items.sort(key=lambda a: a[2])\n",
+    "\n",
     "    print\n",
     "    print \"#\" * 80\n",
-    "    print \"%s WER: %f\" % (caption, mean)\n",
-    "    if len(items) > 0:\n",
-    "        # Filter out all items with WER=0\n",
-    "        items = [a for a in items if a[2] > 0]\n",
-    "        # Order the remaining items by their loss (lowest loss on top)\n",
-    "        items.sort(key=lambda a: a[3])\n",
-    "        # Take only the first 10 items\n",
-    "        items = items[:10]\n",
-    "        # Order this top ten items by their WER (lowest WER on top)\n",
-    "        items.sort(key=lambda a: a[2])\n",
-    "        for a in items:\n",
-    "            print \"-\" * 80\n",
-    "            print \" - WER:    %f\" % a[2]\n",
-    "            print \" - loss:   %f\" % a[3]\n",
-    "            print \" - source: \\\"%s\\\"\" % a[0]\n",
-    "            print \" - result: \\\"%s\\\"\" % a[1] \n",
+    "    print \"%s WER: %f\" % (caption, mean_wer)\n",
+    "    for a in items:\n",
+    "        print \"-\" * 80\n",
+    "        print \" - WER:    %f\" % a[2]\n",
+    "        print \" - loss:   %f\" % a[3]\n",
+    "        print \" - source: \\\"%s\\\"\" % a[0]\n",
+    "        print \" - result: \\\"%s\\\"\" % a[1] \n",
     "    print \"#\" * 80\n",
-    "    print"
+    "    print\n",
+    "    \n",
+    "    return mean_wer"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Another routine will help collecting partial results for the WER reports. The ```results_tuple``` is composed of an array of the original labels, an array of the corrsponding decodings and an array of the corresponding losses. ```returns``` is built up in a similar way, containing just the unprocessed results of one ```session.run``` call (effectively of one batch). Before splicing them into their corresponding ```results_tuple``` lists, labels and decodings are converted to text. In the case of decodings, for now we just pick the first available path."
+    "Another routine will help collecting partial results for the WER reports. The ```results_tuple``` is composed of an array of the original labels, an array of the corrsponding decodings, an array of the corrsponding distances and an array of the corresponding losses. ```returns``` is built up in a similar way, containing just the unprocessed results of one ```session.run``` call (effectively of one batch). Before splicing them into their corresponding ```results_tuple``` lists, labels and decodings are converted to text. In the case of decodings, for now we just pick the first available path."
```

Comment:
Would be nice to switch around the sentence that starts with Before, so it reads

> Labels and decodings are converted to text before splicing them into their corresponding `results_tuple` lists.